var/spack/repos/builtin/packages/qscintilla/package.py

```diff
@@ -72,92 +70,68 @@ def setup_run_environment(self, env):
-            if module != "sip":
-                module = module.split(".")[0]
-                with working_dir(python_platlib):
-                    with open(os.path.join(module, "__init__.py"), "w") as f:
-                        f.write("from pkgutil import extend_path\n")
-                        f.write("__path__ = extend_path(__path__, __name__)\n")
+        with working_dir(os.path.join(self.stage.source_path, "designer")):
+            # TODO: qmake fails with qt6
+            qmake('designer.pro', 'INCLUDEPATH+=../src')
+            make()
+            makefile = FileFilter("Makefile")
+            makefile.filter(r"\$\(INSTALL_ROOT\)" + self.spec["qt"].prefix, "$(INSTALL_ROOT)")
+            make("install")
+
+    @run_after("install", when='+python')
+    def make_qsci_python(self):
+        if "^py-pyqt5" in self.spec:
+            py_pyqtx = "py-pyqt5"
+            pyqtx = "PyQt5"
+            ftoml = 'pyproject-qt5.toml'
+        elif "^py-pyqt6" in self.spec:
+            py_pyqtx = "py-pyqt6"
+            pyqtx = "PyQt6"
+            ftoml = 'pyproject-qt6.toml'
+
+        with working_dir(join_path(self.stage.source_path, "Python")):
+            cp = which('cp')
+            cp(ftoml, 'pyproject.toml')
+            sip_inc_dir = join_path(self.spec[py_pyqtx].prefix,
+                                    self.spec['python'].package.platlib, pyqtx, 'bindings')
+
+            with open('pyproject.toml', 'a') as tomlfile:
+                tomlfile.write(
+                        '\n[tool.sip.project]\nsip-include-dirs = ["'+str(sip_inc_dir)+'"]\n')
+            mkdirp(os.path.join(self.prefix.share.sip, pyqtx))
+
+            if '^py-pyqt5' in self.spec:
+                # QT += widgets and QT += printsupport need to be added to Qsci.pro file
+                # to be generated via project.py
+                qsciproj = FileFilter(join_path("project.py"))
+                ptrn="super().__init__(project, 'Qsci', qmake_CONFIG=qmake_CONFIG",
+                qsciproj.filter(
+                    ptrn+")",
+                    ptrn+",qmake_QT=['widgets','printsupport'])", string=True)
+            sip_build = Executable(self.spec["py-sip"].prefix.bin.join("sip-build"))
+            sip_build(
+                "--target-dir=" + python_platlib,
+                "--qsci-include-dir=" + self.spec.prefix.include,
+                "--qsci-library-dir=" + self.spec.prefix.lib,
+                "--api-dir=" + self.prefix.share.qsci,
+                "--verbose",
+            )
+
+            makefile = FileFilter(join_path("build", "Qsci", "Makefile"))
+            makefile.filter("$(INSTALL_ROOT)", "", string=True)
+            make("install", "-C", join_path("build", "Qsci"), parallel=False)
+
+            makefile = FileFilter(join_path("build", "Makefile"))
+            makefile.filter("$(INSTALL_ROOT)", "", string=True)
+            make("install", "-C", "build/", parallel=False)
```

Comment:
None of these build in parallel?

Reply:
they do build in parallel in fact. It's now fixed.